File: src/main/java/org/jcodec/codecs/h264/io/model/HRDParameters.java
package org.jcodec.codecs.h264.io.model;

/**
 * This class is part of JCodec ( www.jcodec.org )
 * This software is distributed under FreeBSD License
 * 
 * @author The JCodec project
 *
 */
public class HRDParameters {

    // cpb_cnt_minus1
    public int cpbCntMinus1;
    // bit_rate_scale
    public int bitRateScale;
    // cpb_size_scale
    public int cpbSizeScale;
    // bit_rate_value_minus1
    public int[] bitRateValueMinus1;
    // cpb_size_value_minus1
    public int[] cpbSizeValueMinus1;
    // cbr_flag
    public boolean[] cbrFlag;
    // initial_cpb_removal_delay_length_minus1
    public int initialCpbRemovalDelayLengthMinus1;
    // cpb_removal_delay_length_minus1
    public int cpbRemovalDelayLengthMinus1;
    // dpb_output_delay_length_minus1
    public int dpbOutputDelayLengthMinus1;
    // time_offset_length
    public int timeOffsetLength;

}

File: src/main/java/org/jcodec/codecs/h264/io/model/HRDParameters.kt
package org.jcodec.codecs.h264.io.model

/**
 * This class is part of JCodec ( www.jcodec.org )
 * This software is distributed under FreeBSD License
 *
 * @author The JCodec project
 */
class HRDParameters {
    // cpb_cnt_minus1
    @JvmField
    var cpbCntMinus1 = 0

    // bit_rate_scale
    @JvmField
    var bitRateScale = 0

    // cpb_size_scale
    @JvmField
    var cpbSizeScale = 0

    // bit_rate_value_minus1
    @JvmField
    var bitRateValueMinus1: IntArray? = null

    // cpb_size_value_minus1
    @JvmField
    var cpbSizeValueMinus1: IntArray? = null

    // cbr_flag
    @JvmField
    var cbrFlag: BooleanArray? = null

    // initial_cpb_removal_delay_length_minus1
    @JvmField
    var initialCpbRemovalDelayLengthMinus1 = 0

    // cpb_removal_delay_length_minus1
    @JvmField
    var cpbRemovalDelayLengthMinus1 = 0

    // dpb_output_delay_length_minus1
    @JvmField
    var dpbOutputDelayLengthMinus1 = 0

    // time_offset_length
    @JvmField
    var timeOffsetLength = 0
}